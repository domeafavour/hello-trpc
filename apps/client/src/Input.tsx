export type InputProps = {
  name: string;
  value: string;
  disabled?: boolean;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
};

export function Input({ name, value, disabled, onChange }: InputProps) {
  return (
    <input
      name={name}
      className="border-2 border-gray-300 rounded-md p-2 w-1/2"
      disabled={disabled}
      value={value}
      onChange={onChange}
    />
  );
}
