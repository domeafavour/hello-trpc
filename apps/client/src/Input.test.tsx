import { useState } from 'react';
import renderer, { act } from 'react-test-renderer';
import { Input } from './Input';

it('renders Input correctly', () => {
  const tree = renderer
    .create(<Input name="test-input" value="" onChange={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('updates on change', () => {
  function InputWrapper() {
    const [value, setValue] = useState('');
    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
      setValue(event.target.value);
    };

    return <Input name="test-input" value={value} onChange={handleChange} />;
  }

  const instance = renderer.create(<InputWrapper />).root;

  act(() => {
    instance
      .findByType(Input)
      .props.onChange({ target: { value: 'Hello World' } });
  });

  expect(instance.findByType('input').props.value).toBe('Hello World');
});
