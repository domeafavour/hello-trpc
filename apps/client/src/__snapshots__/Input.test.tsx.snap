// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Input correctly 1`] = `
<input
  className="border-2 border-gray-300 rounded-md p-2 w-1/2"
  name="test-input"
  onChange={[Function]}
  value=""
/>
`;
